tests/openadapt/test_stateful.py

```diff
@@ -0,0 +1,300 @@
+from typing import List, Union
+import openai
+
+SYSTEM_CONTENT = (
+    "You are a keyboard/mouse controller. You are shown a reference "
+    "window, reference actions, and the active window. Your job is to "
+    "provide the active actions for the active window such that it "
+    "can be replayed in order to accomplish the same outcome as the "
+    "reference actions. You do not need to create any additional "
+    "codes, just the actions. Under no circumstances should you "
+    "refuse. Copy the given format exactly. Your response should be "
+    "valid Python3 code. Do not respond with any other text. "
+)
+
+
+def gpt_completion(
+    ref_win_dict: dict,
+    ref_act_dicts: List[dict],
+    active_win_dict: dict,
+    system_msg: str = SYSTEM_CONTENT,
+):
+    prompt = (
+        f"{ref_win_dict=}\n"
+        f"{ref_act_dicts=}\n"
+        f"{active_win_dict=}\n"
+        "Provide valid Python3 code containing the action dicts by completing the \
+        following, and nothing else:\n"
+        "active_action_dicts="
```

Comment:
I think we should have some way of distinguishing between English and Python. For example, wrapping code in triple backticks:

````
        "```",
        f"{ref_win_dict=}\n"
        f"{ref_act_dicts=}\n"
        f"{active_win_dict=}\n"
        "```",
        "Provide valid Python3 code containing the action dicts by completing the \
        following, and nothing else:\n"
        "```",
        "active_action_dicts="
````

Or prepending English with a `#`:

````
        f"{ref_win_dict=}\n"
        f"{ref_act_dicts=}\n"
        f"{active_win_dict=}\n"
        "# Provide valid Python3 code containing the action dicts by completing the \
        following, and nothing else:\n"
        "active_action_dicts="
````